src/main/resources/avro/variants.avdl

```diff
@@ -211,9 +273,59 @@ record Variant {
 
   /**
   The variant calls for this particular variant. Each one represents the
-  determination of genotype with respect to this variant.
+  determination of genotype with respect to this variant. `Call`s in this array
+  are implicitly associated with this `Variant`.
+
+  Calls are also available through the `searchCalls()` API method.
+  */
+  union { null, array<Call> } calls = null;
+}
+
+
+/**
+`Allele`s are the key component of the "allelic" model of variation: they are
+single objects that are present in some copy number.
+
+An `Allele` is a contigous piece of sequence that we will want to say is present
+(in general at some copy number) or absent in a sample. Very often it will just
+be a single `Segment` covering the entirety of a novel sequence, but it is also
+common for it to be a mix of `Segment`s on reference and novel sequences, or in
+general to be any contiguous path through the augmented sequence graph.
+
+`Allele`s belong to `VariantSet`s. `Allele`s can also be used to represent the
+reference and alternate alleles of `Variant`s.
+
+Note that `Path`s cannot follow adjacencies not represented in the augmented
+sequence graph. For example, if  an `Allele` that spans a novel deletion is
+required, a new empty-string sequence should exist to describe that deletion
+adjacency, with a `Segment` describing that sequence available through
+`searchVariantSetSequences()` or `getVariantSetSequence()`.
+*/
+record Allele {
+  /**
+  The ID of this `Allele`. If this `Allele` is one `Segment` consisting of
+  the entirety of a sequence, this is equal to the ID of that sequence.
+  */
+  string id;
+
+  /** The ID of the variant set this allele belongs to. */
+  string variantSetId;
+
+  /**
+  The time at which this `Allele` was last updated in
+  milliseconds from the epoch.
+  */
+  union { null, long } updated = null;
```

Comment:
Do we need this at the moment? It seems like keeping an update time is to solve a problem we're a long way from encountering, and can be easily added in later if/when we need it.